File: src/2025-02-24-deepclaude-rust/cpp/exercises/practice/kindergarten-garden/kindergarten_garden.cpp
#include "kindergarten_garden.h"
#include <vector>
#include <string>
#include <algorithm>

namespace kindergarten_garden {

std::array<Plants, 4> plants(const std::string& diagram, const std::string& student) {
    // Declare the list of student names in alphabetical order
    const std::vector<std::string> students = {
        "Alice", "Bob", "Charlie", "David",
        "Eve", "Fred", "Ginny", "Harriet",
        "Ileana", "Joseph", "Kincaid", "Larry"
    };
    
    // Find the student's index
    auto it = std::find(students.begin(), students.end(), student);
    int student_index = std::distance(students.begin(), it);
    
    // Split the diagram into rows
    size_t newline_pos = diagram.find('\n');
    std::string row1 = diagram.substr(0, newline_pos);
    std::string row2 = diagram.substr(newline_pos + 1);
    
    // Get the starting position for this student's plants (2 per row)
    int start_pos = student_index * 2;
    
    // Map the plant codes to Plants enum values
    auto char_to_plant = [](char c) -> Plants {
        switch (c) {
            case 'V': return Plants::violets;
            case 'R': return Plants::radishes;
            case 'C': return Plants::clover;
            case 'G': return Plants::grass;
            default: return Plants::grass; // Fallback, should not happen
        }
    };
    
    // Return the 4 plants for this student
    return {
        char_to_plant(row1[start_pos]),     // First row, first plant
        char_to_plant(row1[start_pos + 1]), // First row, second plant
        char_to_plant(row2[start_pos]),     // Second row, first plant
        char_to_plant(row2[start_pos + 1])  // Second row, second plant
    };
}

}  // namespace kindergarten_garden

File: src/2025-02-24-deepclaude-rust/java/exercises/practice/kindergarten-garden/src/main/java/KindergartenGarden.java
import java.util.List;
import java.util.ArrayList;

class KindergartenGarden {
    enum Plant {
        GRASS('G'),
        CLOVER('C'),
        RADISH('R'),
        VIOLET('V');

        private final char code;

        Plant(char code) {
            this.code = code;
        }

        static Plant fromCode(char code) {
            switch(code) {
                case 'G': return GRASS;
                case 'C': return CLOVER;
                case 'R': return RADISH;
                case 'V': return VIOLET;
                default: throw new IllegalArgumentException("Unknown code: " + code);
            }
        }
    }

    private static final List<String> STUDENTS = List.of(
        "Alice", "Bob", "Charlie", "David",
        "Eve", "Fred", "Ginny", "Harriet",
        "Ileana", "Joseph", "Kincaid", "Larry"
    );

    private final List<String> topRow;
    private final List<String> bottomRow;

    KindergartenGarden(String garden) {
        String[] rows = garden.split("\n");
        if (rows.length != 2) {
            throw new IllegalArgumentException("Garden must have exactly two rows");
        }
        this.topRow = splitIntoChunks(rows[0], 2);
        this.bottomRow = splitIntoChunks(rows[1], 2);
    }

    private List<String> splitIntoChunks(String s, int chunkSize) {
        List<String> chunks = new ArrayList<>();
        for (int i = 0; i < s.length(); i += chunkSize) {
            int end = Math.min(i + chunkSize, s.length());
            chunks.add(s.substring(i, end));
        }
        return chunks;
    }

    List<Plant> getPlantsOfStudent(String student) {
        int studentIndex = STUDENTS.indexOf(student);
        if (studentIndex == -1) {
            throw new IllegalArgumentException("Student not found: " + student);
        }
        List<Plant> plants = new ArrayList<>();
        String topChunk = topRow.get(studentIndex);
        String bottomChunk = bottomRow.get(studentIndex);
        for (char c : topChunk.toCharArray()) {
            plants.add(Plant.fromCode(c));
        }
        for (char c : bottomChunk.toCharArray()) {
            plants.add(Plant.fromCode(c));
        }
        return plants;
    }
}
